includes/server/llms-rest-server-functions.php

```diff
@@ -80,14 +80,125 @@ function llms_rest_server_error( $message = '' ) {
  *
  * @since 1.0.0-beta.9
  *
  * @param array $instructors Array of instructors id.
- *
  * @return boolean
  */
 function llms_validate_instructors( $instructors ) {
 	return ! empty( $instructors ) ? count( array_filter( array_map( 'get_userdata', $instructors ) ) ) === count( $instructors ) : false;
 }
+
+/**
+ * Validate strict positive integer number
+ *
+ * @since [version]
+ *
+ * @param integer $number Integer number to validate.
+ * @return boolean
+ */
+function llms_rest_validate_strictly_positive_int( $number ) {
+	return llms_rest_validate_positive_int( $number, false );
+}
+
+/**
+ * Validate positive integer number including zero
+ *
+ * @since [version]
+ *
+ * @param integer $number Integer number to validate.
+ * @return boolean
+ */
+function llms_rest_validate_positive_int_w_zero( $number ) {
+	return llms_rest_validate_positive_int( $number );
+}
+
+
+/**
+ * Validate positive integer number
+ *
+ * @since [version]
+ *
+ * @param integer $number Integer number to validate.
+ * @param boolean $include_zero Optional. Whether or not 0 is included. Default is `true`.
+ * @return boolean
+ */
+function llms_rest_validate_positive_int( $number, $include_zero = true ) {
+	return false !== filter_var(
+		$number,
+		FILTER_VALIDATE_INT,
+		array(
+			'options' => array(
+				'min_range' => $include_zero ? 0 : 1,
+			),
+		)
+	);
+}
+
+/**
+ * Validate strict positive float number
+ *
+ * @since [version]
+ *
+ * @param integer $number Float number to validate.
+ * @return boolean
+ */
+function llms_rest_validate_strictly_positive_float( $number ) {
+	return llms_rest_validate_positive_float( $number, false );
+}
+
+/**
+ * Validate strict positive float number including zero
+ *
+ * @since [version]
+ *
+ * @param integer $number Float number to validate.
+ * @return boolean
+ */
+function llms_rest_validate_positive_float_w_zero( $number ) {
+	return llms_rest_validate_positive_float( $number );
+}
+
+/**
+ * Validate strict positive float number
+ *
+ * @since [versoin]
+ *
+ * @param integer $number Float number to validate.
+ * @param boolean $include_zero Optional. Whether or not 0 is included. Default is `true`.
+ * @return boolean
+ */
+function llms_rest_validate_positive_float( $number, $include_zero = true ) {
+	// min_range and max_range options for FILTER_VALIDATE_FLOAT are only available since PHP 7.4.
+	$is_float = false !== filter_var( (float) $number, FILTER_VALIDATE_FLOAT );
+	return $is_float && ( $include_zero ? $number >= 0 : $number > 0 );
+}
+
+
+/**
+ * Validate submitted array of integers is an array of real memberships id, or empty.
+ *
+ * @since [version]
+ *
+ * @param array $memberships Array of instructors id.
+ * @return boolean
+ */
+function llms_rest_validate_memberships( $memberships ) {
+
+	$valid = true;
+
+	if ( ! empty( $memberships ) ) {
+		$real_memberships = array_map(
+			function( $membership ) {
+				return is_a( llms_get_post( $membership ), 'LLMS_Membership' );
```

Comment:
I have not run any benchmarks on this but I believe that using a WP Core method here will be slightly more performant than using our post model class. I might be wrong about this, though.

```suggestion
				return ( 'llms_membrship'  === get_post_type( $membership ) );
```